system/src/system_cloud_connection_posix.cpp

```diff
@@ -314,8 +319,27 @@ int system_cloud_disconnect(int flags)
 int system_cloud_send(const uint8_t* buf, size_t buflen, int flags)
 {
     (void)flags;
-    LOG_DEBUG(TRACE, "TX s_state.socket %d buflen %d", s_state.socket, buflen);
-    int r = sock_send(s_state.socket, buf, buflen, 0);
+    int r = 0;
+
+    auto netIfIndex = particle::system::ConnectionManager::instance()->getCloudConnectionNetwork();
+    if (netIfIndex) {
+        if_t ifHandle = 0;
+        if_get_by_index(netIfIndex, &ifHandle);
+        unsigned int ifFlags = 0;
+        if_get_flags(ifHandle, &ifFlags);
+        LOG_DEBUG(TRACE, "Bound cloud netif idx %d flags 0x%0x (IFF_RUNNING %d)", netIfIndex, ifFlags, (ifFlags & IFF_RUNNING)!=0);
+
+        if (!(ifFlags & IFF_RUNNING)) {
+            LOG(WARN, "Netif %d not running. if flags %0x", netIfIndex, ifFlags);
+            r = ERR_CONN;     
+        }
+    }
+    
+    if (!r) {
+        LOG_DEBUG(TRACE, "TX s_state.socket %d buflen %d", s_state.socket, buflen);
+        r = sock_send(s_state.socket, buf, buflen, 0);
+    }
```

Comment:
Unclear if this approach detects underlying network state faster, the netif flags did not seem to change before the wifi client realized the AP had been disconnected:
```
0000050622 [comm.coap] TRACE: Received CoAP message
0000050650 [comm.coap] TRACE: CON POST /E/particle/device/updates/pending size=47 token=02 id=33128
0000050706 [comm.coap] TRACE: Sending CoAP message
0000050736 [comm.coap] TRACE: ACK 0.00  size=4 token= id=33128
0000050770 [system] TRACE: Bound cloud netif idx 5 flags 0x18043 (IFF_RUNNING 1).    // <--- flags = 0x18043 when IF is connected
0000050817 [system] TRACE: TX s_state.socket 0 buflen 53
0000050949 [system] TRACE: RX s_state.socket 0 buflen 1485 recvd 33
0000050987 [comm.coap] TRACE: Received CoAP message
0000051021 [comm.coap] TRACE: ACK 0.00  size=4 token= id=10
0000051052 [system] INFO: All handshake messages have been processed
0000051194 [system] INFO: Cloud connected
0000051219 [system.ledger] TRACE: Connected

// < WIFI AP POWERED OFF >

0000076144 [comm.coap] TRACE: Sending CoAP message
0000076172 [comm.coap] TRACE: CON 0.00  size=4 token= id=11
0000076200 [system] TRACE: Bound cloud netif idx 5 flags 0x18043 (IFF_RUNNING 1) // <---- netif flags remain unchanged
0000076239 [system] TRACE: TX s_state.socket 0 buflen 53
0000079294 [ncp.rltk.client] INFO: disconnected       // <---- rtlk NCP client detects AP is gone, notifies network manager
0000079296 [ncp.rltk.client] TRACE: NCP connection state changed: 0
0000079297 [net.rltkncp] TRACE: NCP event 2
0000079298 [net.rltkncp] TRACE: State changed event: 0
0000079299 [net.ifapi] INFO: Netif wl4 link DOWN, profile=NONE
0000079301 [net.rltkncp] ERROR: linkOutput up=1 link_up=0
0000079302 [net.ifapi] TRACE: Netif wl4 ipv4 configuration changed
0000079305 [system.nm] INFO: State changed: IP_CONFIGURED -> IFACE_LINK_UP
0000079306 [system.nm] WARN: Cloud connection interface 5 link state down, disconnecting
```